### PR TITLE
Improvements in MEMO in withdraw funds

### PR DESCRIPTION
**Pull request by BartoszBilejczyk**

### CODE REVIEW

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html [hunk 1]
```diff
@@ -121,29 +120,36 @@
...
-          Total Amount:
+          Total Withdraw Amount:
           <span class="total-amount">
-            {{ totalWithdrawAmount }}
+            {{ withdrawFundsForm.get('amount').value + withdrawFundsForm.get('amount').value * 0.002 | adsharesTokenValue }}
           </span>
         </p>
         <p>
-          Left: {{ fundsLeft }}
+          Left: {{ financialData.totalFunds - withdrawFundsForm.get('amount').value + withdrawFundsForm.get('amount').value * 0.002 | adsharesTokenValue }}
```
Comment (KonradFak) [inline src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html]: Are you sure of this calculation?

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html [hunk 2]
```diff
@@ -154,29 +160,57 @@
           adsh-btn--white
           adsh-btn--block
           row
-          align-center
-          adsh-funds-summary__deposit-funds"
+          align-center"
+        (click)="toggleMemoInput(!memoInputActive); $event.preventDefault()"
```
Comment (KonradFak) [inline src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html]: We should avoid multiple operations in html attributes. If `$event.preventDefault()` is necessary it should be moved to `toggleMemoInput`

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html [hunk 3]
```diff
@@ -154,29 +160,57 @@
...
       >
         <img
           alt="Credit Card Icon"
           class="
-            adsh-icon
-            adsh-icon--prepend"
+          adsh-icon
+          adsh-icon--prepend"
```
Comment (KonradFak) [inline src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html]: Fix indentation

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html [hunk 4]
```diff
@@ -154,29 +160,57 @@
...
           src="/assets/images/plus-circle--blue.svg"
         >
         <span
           class="
-            adsh-copy
-            adsh-copy--x-dark"
+          adsh-copy
+          adsh-copy--x-dark
+          adsh-copy--semi"
```
Comment (KonradFak) [inline src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html]: Fix indentation

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html [hunk 5]
```diff
@@ -154,29 +160,57 @@
...
+          for="address"
+          class="adsh-form-label"
+        >
+          MEMO
+        </label>
+        <input
+          id="memo"
+          class="
+          adsh-form-input
+          adsh-form-input--full"
```
Comment (KonradFak) [inline src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html]: Fix indentation

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.ts [hunk 6]
```diff
@@ -32,28 +31,35 @@ export class WithdrawFundsDialogComponent extends HandleSubscription implements
...
 
   ngOnInit() {
-    this.createForm();
-
-    const userEthAddressSubscription = this.store.select('state', 'user', 'data', 'financialData', 'userEthAddress')
-      .subscribe((userEthAddress: string) => {
-        this.userEthAddress = userEthAddress;
+    const userEthAddressSubscription = this.store.select('state', 'user', 'data', 'financialData')
+      .subscribe((financialData: UserFinancialData) => {
+        this.financialData = financialData;
```
Comment (KonradFak) [inline src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.ts]: Subscribe method should fit one line

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.ts [hunk 7]
```diff
@@ -32,28 +31,35 @@ export class WithdrawFundsDialogComponent extends HandleSubscription implements
...
         Validators.required,
         Validators.minLength(42),
         Validators.maxLength(42)
       ]),
       amount: new FormControl('', [
         Validators.required
+      ]),
+      memo: new FormControl('', [
+        Validators.maxLength(32)
       ])
```
Comment (KonradFak) [inline src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.ts]: declarations `amount` and `memo ` would fit one line

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html (hunk 3 above)
Comment (BartoszBilejczyk) [reply src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html]: Done

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html (hunk 4 above)
Comment (BartoszBilejczyk) [reply src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html]: Done

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html (hunk 5 above)
Comment (BartoszBilejczyk) [reply src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html]: Done

File: src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html (hunk 2 above)
Comment (BartoszBilejczyk) [reply src/app/common/dialog/withdraw-funds-dialog/withdraw-funds-dialog.component.html]: Done

**Pull request merged**